### PR TITLE
Guard the boundary marking code for exodus files.

### PR DESCRIPTION
**Pull request by juan42**
Loading an exodus mesh file resulted in a memory error in PETSc. This fixes issue #995.

### CODE REVIEW

**Comment by wence-** [comment]
Can you provide a small (couple of elements) exodus mesh that we can add to the test suite so we can test this code path?

**Comment by juan42** [comment]
Sure. I will add a test including a small exodus mesh from a recent cubit version.

**Comment by wence-** [comment]
such test data lives in `tests/meshes/` right now.

File: tests/regression/test_exodus_mesh.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import pytest
```
Comment (thomasgibson) [inline tests/regression/test_exodus_mesh.py]: Add `from __future__ import absolute_import, print_function, division` to satisfy the flake8 requirements

**Comment by juan42** [comment]
The tests on travis/drone fail because 
```
E   [0] This method requires ExodusII support. Reconfigure using --download-exodusii
```

File: tests/regression/test_exodus_mesh.py [hunk 2]
```diff
@@ -8,7 +8,10 @@
 
 @pytest.fixture
 def exodus_mesh():
-    return Mesh(join(cwd, "..", "meshes", "brick.e"))
+    try:
+        return Mesh(join(cwd, "..", "meshes", "brick.e"))
+    except PETSc.Error as e:
+        assert e.ierr == 56
```
Comment (miklos1) [inline tests/regression/test_exodus_mesh.py]: You should rather make an xfail in this case.

**Comment by wence-** [comment]
Fixed the petsc import and merged.

**Comment by wence-** [comment]
Thanks!

**Pull request merged**